### PR TITLE
Fix x32 build

### PR DESCRIPTION
**Pull request by coslyk**
I copy the patches from the ubuntu's deb packages here. Maybe it can fix the builds on x32 and ARM?

### CODE REVIEW

**Comment by TingPing** [comment]
bot, build

**Comment by coslyk** [comment]
It seems that ARM build is fixed, build fails on 32bit ARM just because there is no enough memory in the builder:
```
fatal error: lib/libQt5WebKit.so.5.212.0: mmap: failed to allocate 2809472704 bytes for output file: Cannot allocate memory
```

**Comment by TingPing** [comment]
Hmm that translates to 2.6GB, surely that could be possible?

**Comment by coslyk** [comment]
> Hmm that translates to 2.6GB, surely that could be possible?

I think that is the GCC's RAM usage during linking?

**Comment by coslyk** [comment]
The build failed on i386 fails maybe because the macro `JSC_HOST_CALL` is not set correctly. The error output is:
```
../Source/WebCore/bindings/js/JSDOMConstructor.h:195:35: error: invalid conversion from ‘JSC::EncodedJSValue (*)(JSC::ExecState*)’ {aka ‘long long int (*)(JSC::ExecState*)’} to ‘JSC::NativeFunction’ {aka ‘long long int (__attribute__((fastcall)) *)(JSC::ExecState*)’} [-fpermissive]
     constructData.native.function = construct;
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
```

`construct` is declared as 
```
static JSC::EncodedJSValue JSC_HOST_CALL construct(JSC::ExecState*);
```
`JSC_HOST_CALL` seems not to be correct. It's defined in `webkit/Source/WTF/wtf/Platform.h`:
```
#if CPU(X86) && COMPILER(MSVC)
#define JSC_HOST_CALL __fastcall
#elif CPU(X86) && COMPILER(GCC_OR_CLANG)
#define JSC_HOST_CALL __attribute__ ((fastcall))
#else
#define JSC_HOST_CALL
#endif
```
Maybe some changes in the Freedesktop SDK cause this error?

File: io.qt.qtwebkit.BaseApp.json [hunk 1]
```diff
@@ -22,6 +22,30 @@
           "type": "archive",
           "url": "https://github.com/qt/qtwebkit/archive/v5.212.0-alpha2.tar.gz",
           "sha256": "6db43b931f64857cfda7bcf89914e2730b82164871a8c24c1881620e6bfdeca1"
+        },
+        {
+          "type": "patch",
+          "path": "patches/detect_opengl_es2.diff"
```
Comment (eszlari) [inline io.qt.qtwebkit.BaseApp.json:28]: You can put all patches in one source element if you use `paths` instead of `path`.

**Comment by coslyk** [comment]
Try this new patch?

**Comment by TingPing** [comment]
bot, build

**Comment by coslyk** [comment]
I filed an issue in upstream for help: https://github.com/annulen/webkit/issues/795

**Pull request merged**